.github/workflows/main.yml

```diff
@@ -46,52 +42,54 @@ jobs:
+            if [[ "$REPO_HEAD" == "AbsaOSS/libvcx" ]]
             then
-              DOCKER_TAG_VERSION="$BRANCH_NAME"
-              echo "Running push on master but version was not bumped. Will NOT release. The latest release version is $REPO_VERSION_HIGHEST."
+              echo "This is not fork."
+              IS_FORK="false";
             else
-              RELEASE_VERSION="$REPO_VERSION_DESIRED"
-              DOCKER_TAG_VERSION="$REPO_VERSION_DESIRED"
-              echo "Running push on master, version was bumped $REPO_VERSION_HIGHEST->$RELEASE_VERSION. Will release $RELEASE_VERSION."
-            fi
+              echo "This is fork. "
+              IS_FORK="true";
+            fi;
           else
-             FULL_REPO_SLUG=`echo "${{github.event.pull_request.head.repo.full_name}}" | sed "s/[^[:alnum:]-]//g" | tr '[:upper:]' '[:lower:]'` # lowercase, only alphanumeric and dash
-             echo ::set-env name=FULL_REPO_SLUG::$(echo $FULL_REPO_SLUG)
-             DOCKER_TAG_VERSION="$FULL_REPO_SLUG-$BRANCH_NAME"
-             echo "This is not push to master. Will NOT release. The latest release version is $REPO_VERSION_HIGHEST."
-          fi
+            echo "Not PR so this is not fork."
+            IS_FORK="false";
+          fi;
 
-          echo ::set-env name=DOCKER_TAG_VERSION::$(echo $DOCKER_TAG_VERSION)
+          if [[ "$IS_FORK" == "false" ]]
+          then
+            PUBLISH_VERSION="$REPO_VERSION_HIGHEST-$BRANCH_NAME-${{ github.run_number }}"
+            echo "This is not push to master. Will be publishing version: $PUBLISH_VERSION"
+          else
+            echo "This PR is from fork, nothing will be published because the CI wouldn't be able to access repo secrets to perform publish."
+            PUBLISH_VERSION=""
+          fi;
 
           LIBVCX_SOURCE_HASH=${{ hashFiles('libvcx') }}
           LIBVCX_DOCKERFILE_HASH=${{ hashFiles('ci/libvcx.dockerfile') }}
+          LIBVCX_NODE_WRAPPERS_HASH=${{ hashFiles('wrappers/node') }}
           JAVA_WRAPPERS_HASH=${{ hashFiles('wrappers/java') }}
           POOL_DOCKERFILE_HASH=${{ hashFiles('ci/indy-pool.dockerfile')}}
 
-          LIBVCX_HASH=${LIBVCX_SOURCE_HASH:0:15}-${LIBVCX_DOCKERFILE_HASH:0:15}
+          LIBVCX_HASH=${LIBVCX_SOURCE_HASH:0:15}-${LIBVCX_NODE_WRAPPERS_HASH:0:15}-${LIBVCX_DOCKERFILE_HASH:0:15}
           ANDROID_HASH=${LIBVCX_SOURCE_HASH:0:15}-${JAVA_WRAPPERS_HASH:0:15}
           POOL_HASH=${POOL_DOCKERFILE_HASH:0:15}
 
-          echo ::set-env name=LIBVCX_HASH::$(echo $LIBVCX_HASH)
-          echo ::set-env name=ANDROID_HASH::$(echo $ANDROID_HASH)
-          echo ::set-env name=POOL_HASH::$(echo $POOL_HASH)
-
-          echo ::set-env name=CACHE_KEY_LIBVCX::$BRANCH_NAME-$LIBVCX_HASH
-          echo ::set-env name=CACHE_KEY_ANDROID::$BRANCH_NAME-$ANDROID_HASH
-          echo ::set-env name=CACHE_KEY_POOL::$BRANCH_NAME-$POOL_HASH
+          echo ::set-env name=PUBLISH_VERSION::$(echo $PUBLISH_VERSION)
```

Comment:
No need to echo here, the value will be present multiple times in the logs already.